### PR TITLE
Add filter option being able to modify routes based on req or hostname

### PR DESCRIPTION
**Pull request by dohomi**
With this PR the module options are extended with a filter being able to filter the routes based on request or hostname.

This is useful if a project is hosted on multiple top-level-domain and each represents one language.

Hope you find this PR useful. I'm not fully aware if it works through caching - maybe the cache mechanism to make this work needs to be set to 0 for no caching.

### CODE REVIEW

**Comment by dohomi** [comment]
@NicoPennec could you have a look at this PR? would be great because I think it solves a couple of issues due to the nature of modifying the routes array before render. Thanks

**Comment by dohomi** [comment]
@NicoPennec is it possible to review this PR? I think its quite helpful (I use it currently for my deployments)

**Comment by dohomi** [comment]
@Atinux @NicoPennec it has been a while. Any chance getting this merged? Thanks!

**Comment by manniL** [comment]
@dohomi Currently there are merge conflicts present.

**Comment by dohomi** [comment]
@manniL thanks I just resolved them

File: src/index.js [hunk 1]
```diff
@@ -155,6 +155,11 @@ function createSitemap (options, routes, req) {
   sitemapConfig.hostname = options.hostname ||
     (req && `${isHTTPS(req) ? 'https' : 'http'}://${req.headers.host}`) || `http://${hostname()}`
 
+  // option to filter on each sitemap request over submitted routes
+  if (typeof options.filter === 'function') {
+    routes = options.filter({routes, hostname: sitemapConfig.hostname, req})
```
[the remark this answers is not in the document]
Comment (dohomi) [reply src/index.js]: @NicoPennec I changed the lines according to your comments. 

Renamed to postProcess and changed the submitted arguments.

I'm very flexible with the naming, just let me know what suits you best.

**Comment by NicoPennec** [comment]
feat: add custom `filter` option

**Pull request merged**